### ISSUE TITLE
Add Symbol for providing a friendly name

### ISSUE THREAD

**Issue by phillipskevin** [canjs/can-symbol#16]
We need to create a Symbol that can be used to retrieve a "friendly" name to be used in log messages. For example, this would give `PersonVM(12)` so that it can be used in the `log()` messages in the following code:

```js
var ViewModel = DefineMap.extend("PersonVM",{
  get fullName() {
    return this.first + " " + this.last;
  },
  first: "string",
  last: "string"
});
var vm = new ViewModel();

vm.log();

// PersonVM(12).first <"Payal">
// PersonVM(12).last <"Shah">
// PersonVM(12).fullName <"Payal Shah">
```

This should default to `<constructor name>(<cid>)`.

This should be added to:
- [ ] DefineMap
- [ ] DefineList
- [ ] Observation
- [ ] Compute

Eventually, it should be added to:
- [ ] Scopes
- [ ] Stache expressions
- [ ] Route
- [ ] can-connect instance and list stores

**Comment by matthewp** [comment]
So, this comes through `.extend()`, right? In that case implementing this in can-construct should fix it everywhere I would think.

**Comment by phillipskevin** [comment]
Yeah, I think for the initial list that will cover it. There will be more work to get this working with Scopes, Stache Expressions, etc.

**Comment by m-mujica** [comment]
so, to implement this, I'll need to: 

- [ ] - Add the symbol [around here](https://github.com/canjs/can-symbol/blob/6682c11b8f66065ac20ae91d902fb5ed173a3fcb/can-symbol.js#L81), any idea on the name? `friendlyName`?  just `name` so it is Symbol.for("can.name")`?

- [ ] - with can-reflect and can-symbol, implement the actual behavior [around here](https://github.com/canjs/can-construct/blob/a43b74d811330ca9b01e576a1c3c13926f8eacad/can-construct.js#L856)?

```js
canReflect.set(Constructor.prototype, canSymbol.for("can.friendlyName"), function() {
  // I guess there should be some defensive code here to handle missing name/cid properties
  return `${this.name}(${this.cid})`;
})
```

am I missing anything? @phillipskevin?

**Comment by m-mujica** [comment]
I'm going to re-open this until the symbol is implemented in the first items mentioned above.

**Comment by m-mujica** [comment]
So, I added the `can.getName` symbol and released it under `1.3.0`, then created the following PRs: 

- [can-reflect](https://github.com/canjs/can-reflect/pull/62)
- [can-define](https://github.com/canjs/can-define/pull/276)
- [can-observation](https://github.com/canjs/can-observation/pull/117)
- [can-compute](https://github.com/canjs/can-compute/pull/108)

@justinbmeyer after re-reading [your comment](https://github.com/canjs/canjs/issues/3278#issuecomment-335496772) I'm wondering if the symbol implementation in the PRs above is not enough, what do you have in mind to provide the content between `{}`, `[]` and `<>`? 

Because my PRs make it so `cid` is the thing between the brackets. cc: @phillipskevin
